lib/DwarfImporter/DwarfImporter.cpp

```diff
@@ -146,23 +145,21 @@ class DwarfToModelConverter {
     AltIndex(AltIndex),
     DICtx(DICtx) {
 
-    // Detect default ABI from architecture
-    // TODO: this needs to be refined
-    switch (DICtx.getArch()) {
-    case llvm::Triple::x86_64:
-      DefaultABI = model::ABI::SystemV_x86_64;
-      break;
-    default:
-      DefaultABI = model::ABI::Invalid;
-    }
+    // Ensure the architecture is consistent.
+    auto Arch = model::Architecture::fromLLVMArchitecture(DICtx.getArch());
+    revng_assert(Arch == Model->Architecture);
+
+    // Detect default ABI from the architecture.
+    if (Model->DefaultABI == model::ABI::Invalid)
+      Model->DefaultABI = model::ABI::getDefault(Model->Architecture);
```

Comment:
Is this a "nice to have" opportunistic improvement of the model or was it addressing some specific problem?

Reply:
It's because "if the architecture is `x86_64` - set the abi to `SystemV_x86_64`, `Invalid` otherwise" didn't sit right with me. So here's a "Basic" way to set it.

And if you're about the check whether it's invalid, there's a possibility that this model already has a DefaultABI defined (since it's technically an input), I don't think we should ovewrite something like that.